android/src/main/java/com/trikot/sample/MainActivity.kt

```diff
@@ -1,23 +1,21 @@
 package com.trikot.sample
 
-import android.os.Bundle
-import androidx.appcompat.app.AppCompatActivity
-import androidx.databinding.DataBindingUtil
-import androidx.lifecycle.ViewModelProviders
-import com.mirego.trikot.viewmodels.LifecycleOwnerWrapper
-import com.trikot.sample.viewmodels.AndroidAppViewModel
+import com.trikot.sample.common.BaseViewModelActivity
 import com.trikot.sample.databinding.ActivityMainBinding
+import com.trikot.sample.viewmodels.base.BaseNavigationDelegate
+import com.trikot.sample.viewmodels.home.HomeViewModel
+import com.trikot.sample.viewmodels.home.HomeViewModelController
 
-class MainActivity : AppCompatActivity() {
-    lateinit var binding: ActivityMainBinding
-    private lateinit var appViewModel: AndroidAppViewModel
+class MainActivity : BaseViewModelActivity<BaseNavigationDelegate, HomeViewModel>() {
+    override val viewModelController: HomeViewModelController by lazy {
+        getViewModelController(
+            HomeViewModelController::class
+        )
```

Comment:
I always find it odd when a controller get its dependency itself like this, should it be passed in it's constructor instead?

Reply:
You can't do that on Android for serialization support. An activity/fragment must always obtain dependencies itself based on the serializable parameters it receives. Not doing this would mean having serializable ViewModelControllers